src/Pos/Block/Network/Listeners.hs

```diff
@@ -78,24 +79,23 @@ stubListenerConv' = unproxy $ \(_ :: Proxy ssc) ->
 handleGetHeaders
     :: forall ssc m.
        (WorkMode ssc m)
-    => (ListenerSpec m, OutSpecs)
-handleGetHeaders = listenerConv $ \_ peerId conv -> do
-    logDebug $ "handleGetHeaders: request from " <> show peerId
-    handleHeadersCommunication conv
+    => m (ListenerSpec m, OutSpecs)
+handleGetHeaders = reifyMsgLimit (Proxy @MsgGetHeaders) $ \limitProxy ->
```

Comment:
What would you propose?
As far as I see, that would be cool if limit would automatically change once an update arrives.
But now, listeners are formed at node start, and node starter accepts `[Listener]`, not `[m Listener]` or something, so we have no choice but to fixate used limits at the beginning as well.